### PR TITLE
SDK changes for Edit/Delete Threads

### PR DESCRIPTION
**Pull request by vatsa287**

### CODE REVIEW

File: python/kadalu_content_apis/objects.py [hunk 1]
```diff
@@ -225,3 +225,25 @@ def list_shares(self, page=1, page_size=30):
     def share(self, share_id):
         """ Return a Share instance """
         return Share(self.conn, self.folder_name, self.path, share_id)
+
```
Comment (aravindavk) [inline python/kadalu_content_apis/objects.py:228]: Create a new object in this file 

```python
class DocumentThread(Generic):
    def __init__(self, conn=None, thread_id=None, data={}):
        """ Intialise DocumentThread/Object """
        super().__init__(data)
        if conn is not None:
            self.conn = conn
        if thread_id is not None:
            self.thread_id = thread_id

    def update(self, thread_type=None, meta=None, data=None):
        ...

    def delete(self):
        ...
```

File: python/kadalu_content_apis/__init__.py [hunk 2]
```diff
@@ -110,3 +110,11 @@ def list_templates(self, page=1, page_size=30):
     def template(self, name):
         """ Return Template instance """
         return Template(self, "/", name)
+
```
Comment (aravindavk) [inline python/kadalu_content_apis/__init__.py:113]:
```python
def thread(self, thread_id):
    return DocumentThread(self, thread_id)
```

Comment (vatsa287) [reply python/kadalu_content_apis/__init__.py:113]: Updated

File: python/kadalu_content_apis/objects.py (hunk 1 above)
Comment (vatsa287) [reply python/kadalu_content_apis/objects.py:228]: Updated

File: python/kadalu_content_apis/objects.py [hunk 3]
```diff
@@ -225,3 +225,34 @@ def list_shares(self, page=1, page_size=30):
...
+        resp = self.conn.http_put(
+            url,
+            {
+                "data": data,
+                "meta": meta,
+                "type": thread_type
+            }
+        )
+
+        outdata = response_object_or_error(Document, resp, 200)
```
Comment (aravindavk) [inline python/kadalu_content_apis/objects.py:251]: `outdata = response_object_or_error(DocumentThread, resp, 200)`

File: python/kadalu_content_apis/objects.py [hunk 4]
```diff
@@ -225,3 +225,34 @@ def list_shares(self, page=1, page_size=30):
...
+        outdata.conn = self.conn
+        return outdata
+
+    def delete(self):
+        url = f"{self.conn.url}/api/threads/{self.thread_id}"
+        resp = self.conn.http_delete(url)
+        return response_object_or_error(Document, resp, 204)
```
Comment (aravindavk) [inline python/kadalu_content_apis/objects.py:258]: same here.

Comment (vatsa287) [reply python/kadalu_content_apis/objects.py]: updated

File: python/kadalu_content_apis/objects.py (hunk 3 above)
Comment (vatsa287) [reply python/kadalu_content_apis/objects.py]: ack

**Review by aravindavk — APPROVED** [review]

**Pull request merged**